sys/src/cmd/gdbserver/gdbstub.c

```diff
@@ -718,30 +716,30 @@ gdb_cmd_binwrite(struct state *ks)
 		strcpy((char *)remcom_out_buffer, "OK");
 }
 
-/* Handle the 'D' or 'k', detach or kill packets */
+/* Handle the 'D', detach packet */
 static void
-gdb_cmd_detachkill(struct state *ks)
+gdb_cmd_detach(struct state *ks)
 {
 	char *error;
 
-	/* The detach case */
-	if (remcom_in_buffer[0] == 'D') {
-		error = dbg_remove_all_break(ks);
-		if (error < 0) {
-			error_packet(remcom_out_buffer, error);
-		} else {
-			strcpy((char *)remcom_out_buffer, "OK");
-			//connected = 0;
-		}
-		put_packet(remcom_out_buffer);
+	error = dbg_remove_all_break(ks);
+	if (error < 0) {
+		error_packet(remcom_out_buffer, error);
 	} else {
-		/*
-		 * Assume the kill case, with no exit code checking,
-		 * trying to force detach the debugger:
-		 */
-		dbg_remove_all_break(ks);
+		strcpy((char *)remcom_out_buffer, "OK");
 		connected = 0;
+		sendctl(ks->threadid, "start");
 	}
+	put_packet(remcom_out_buffer);
+}
+
+/* Handle the 'k' kill packet */
+static void
+gdb_cmd_kill(struct state *ks)
+{
+	sendctl(ks->threadid, "kill");
+	connected = 0;
+	print("Process %d killed by gdb\n", ks->threadid);
```

Comment:
I think it's useful feedback for the user, especially if the process is not visible.